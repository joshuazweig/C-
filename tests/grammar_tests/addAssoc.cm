int main() {
    int x;

    x = 1 + 2 + 3;
}

