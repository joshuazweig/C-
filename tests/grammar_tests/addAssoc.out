ACCEPT
[program:
  [decls:
    [decls:]
    [fdecl:
      [typ: INT]
      ID
      LPAREN
      [formals_opt:]
      RPAREN
      LBRACE
      [combined_list:
        [combined_list: [combined_list:] [vdecl: [typ: INT] ID SEMI]]
        [stmt:
          [expr:
            ID
            ASSIGN
            [expr:
              [expr: [expr: LITERAL] PLUS [expr: LITERAL]]
              PLUS
              [expr: LITERAL]
            ]
          ]
          SEMI
        ]
      ]
      RBRACE
    ]
  ]
  EOF
]
